node_heap.go

```diff
@@ -53,14 +53,9 @@ func (nh *NodeHeap) TopN(n int) []*Node {
 	m := make([]*Node, 0, n)
 	nh.lk.RLock()
 	defer nh.lk.RUnlock()
-	for i := 0; i < n; i++ {
-		node := heap.Pop(nh)
-		if n, ok := node.(*Node); ok {
-			m = append(m, n)
-		}
-	}
-	for _, n := range m {
-		heap.Push(nh, n)
+	for i := 0; i < n && i < len(nh.Nodes); i++ {
```

Comment:
I thought this returns an unsorted list of candidates?

Reply:
I think if we always use `heap.Push` when adding new elements, then the `nh.Nodes` remains in sorted order?

Reply:
My train of thought is that if the node(s) stats change though after they have been pushed, then `nh.Nodes` doesn't remain sorted because `heap.push` only sorts on insertion. 

Also if the stats of already ordered nodes change, then I think `heap.Push` actually would probably never keep a sorted list because`heap.Push` would only work properly if the heap is correctly sorted when the insertion occurs. 

Not sure if im missing something here maybe?

Reply:
not sure how much churn we expect - hopefully in the steady state not too much.

the answer to changing `Priority` of nodes that lead to an unsorted heap is to use `heap.Init` at the beginning of the call to TopN - or to use `heap.Fix` whenever specific node priorities are changed.

Since we only call `TopN` in the periodic pool refresh, so once every 5 minutes, it's probably no problem to re-establish the heap ordering each time.